Slovak translation updates / improvements from Crowdin

das-g opened the pull request:
made by:

* [vskriv](https://crowdin.com/profile/vskriv)
* Miroslav Bagljas (@mbag / [mbag](https://crowdin.com/profile/mbag))
* Norma Driske (@normade / [normade](https://crowdin.com/profile/normade))
* Tomasz Magulski (@magul / [magul](https://crowdin.com/profile/magul))
* Alexander Kravčík (@freeinko / [freeinko](https://crowdin.com/profile/freeinko))
* Matus Valo ([matusvalo](https://crowdin.com/profile/matusvalo))
* Miroslav Michalička ([miro.michalicka](https://crowdin.com/profile/miro.michalicka))
* [jsicho](https://crowdin.com/profile/jsicho)
* Anna Warzecha (vi) (@aniav / [aniav](https://crowdin.com/profile/aniav))
* [enigma917](https://crowdin.com/profile/enigma917)
* [Adamantike](https://crowdin.com/profile/adamantike)

🥇 and mostly by:
* Ivana Kellyerova (@jenxness / [jenx](https://crowdin.com/profile/jenx))

closes #1767

das-g commented:
@DjangoGirls/sk-pr If by reviewing this PR, you notice that corrections or improvements are needed, please make them on Crowdin. (You can _additionally_ comment here on GitHub if you want, but we need to have the change in Crowdin anyway, to facilitate future updates and to avoid regressions.) Please notify me when I either should update this PR from Crowdin (when you've made changes there) or when you think this PR is ready to be merged (which cause the changes in this PR to be shipped to https://tutorial.djangogirls.org/).

das-g commented:
For the record, this is how I made this PR:

on a checkout of the `master` branch:
1. delete Slovak Markdown files with `git rm 'sk/*.md'`
2. temporarily remove the Slovak directory (which still contains images) with `rm sk`
3. go to https://crowdin.com/project/django-girls-tutorial/sk
4. click the Upload-Download button and choose "Download"
5. unpack the downloaded zip archive
6. from the zip archive content, rename `master/sk-SK` to `sk`
7. move that folder into the repo
8. restore images in the Slovak directory with `git checkout sk`
9. `git add sk`
10. look through output of `git log -- sk` to see when last update from Crowdin was. (Now that I look again, I'm unsure whether I identified the correct commit)
11. go to https://crowdin.com/project/django-girls-tutorial/activity-stream, set language filter to Slovak and check who all contributed since then
12. mention also real name and GitHub account in commit message where I know them, or can be reasonably sure through to matching account names + matching faces in profile images or matching account names + matching real names on profiles
13. commit, push, open PR

jenxness commented:
Thanks a ton @das-g for creating the PR.

I've made a handful of changes on Crowdin so we'd need a PR update, please.

While checking the PR I noticed some translations were outdated (I changed another batch of things a couple of days ago after the initial huge translation round -- those are not included in the Crowdin dump here for whatever reason). Could you please double-check (or I can do that too of course) after the next pull that e.g. the name of the "`practice`" directory in `intro_to_command_line/README.md` that's created somewhere around line 249 is "`mkdir pokus`" instead of "`mkdir practice`"? If that's the case it should be newest version of the translations and can be merged right away.

ekohl commented:
@das-g that looks complicated. This worked for me:
```sh
wget https://crowdin.com/backend/download/project/django-girls-tutorial/sk.zip
unzip sk.zip
find sk -name '*.md' -delete
rsync -av master/sk-SK/* sk/
rm -rf master
git add sk
```

I'm playing with writing a GH action to automate it.

ekohl commented:
> I'm playing with writing a GH action to automate it.

https://github.com/DjangoGirls/tutorial/pull/1769 would be that. It's based on what we do in The Foreman project.

das-g commented:
> While checking the PR I noticed some translations were outdated (I changed another batch of things a couple of days ago after the initial huge translation round -- those are not included in the Crowdin dump here for whatever reason). Could you please double-check (or I can do that too of course) after the next pull that e.g. the name of the "`practice`" directory in `intro_to_command_line/README.md` that's created somewhere around line 249 is "`mkdir pokus`" instead of "`mkdir practice`"? If that's the case it should be the newest version of the translations and can be merged right away.

@magul Any idea how I can get an up-to-date translation out of Crowdin? I've deleted the alternative translation suggestions using "practice" and only kept those with "pokus", but if I update my local branch (this time according to @ekohl's much simpler method at https://github.com/DjangoGirls/tutorial/pull/1768#issuecomment-1240616281), I get no new changes.

jenxness commented:
I also noticed that. Approving a translation doesn't seem to help either (see e.g. in `SUMMARY.md`, in the downloaded translations the chapter about deployment is called "Deploy!" even though I'd changed that a couple of days ago to "Nasadenie!" and even approved it, trying to see if it'd make a difference for downloading). It seems for some reason the download is stuck at an earlier version of the translations and all translations changed after the initial big round are just ignored.

jenxness commented:
@das-g The translations should be up to date if you download again now. I've poked Crowdin support and they said the project needs to be "built" in Crowdin first, which they did for me. So should be ready now.

magul commented:
There’s build (& download) button in their UI and they also expose an API
call for that.

W dniu czw., 8.09.2022 o 19:00 Ivana Kellyerova ***@***.***>
napisał(a):

> @das-g <https://github.com/das-g> The translations should be up to date
> if you download again now. I've poked Crowdin support and they said the
> project needs to be "built" in Crowdin first, which they did for me. So
> should be ready now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DjangoGirls/tutorial/pull/1768#issuecomment-1240980438>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAI4OUX4XMG77STUHY47WH3V5ILSZANCNFSM6AAAAAAQHNJRTA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Tomasz Magulski

das-g commented:
> There’s build (& download) button in their UI

Found it! Funny that I never noticed that before.

> and they also expose an API call for that.

@ekohl That's probably something you'll need for #1769.
